utils/message-generator/src/executor.rs

```diff
@@ -758,19 +759,21 @@ impl Executor {
 
 fn change_fields<'a>(
     m: AnyMessage<'a>,
-    replace_fields: Vec<(String, String)>,
-    values: HashMap<String, String>,
+    replace_fields: Vec<ReplaceField>,
+    values: HashMap<String, serde_json::Value>,
 ) -> AnyMessage<'static> {
     let mut replace_fields = replace_fields.clone();
     let next = replace_fields
         .pop()
         .expect("replace_fields cannot be empty");
-    let keyword = next.1;
-    let field_name = next.0;
+    let keyword = next.keyword;
+    let field_name = next.field_name;
     let value = values
         .get(&keyword)
         .expect("value not found for the keyword");
 
+    let message_as_string = serde_json::to_string(&m.clone()).unwrap();
+
     match m.clone() {
         AnyMessage::Common(m) => {
             let path = match m {
```

Comment:
You do not need the path string. You can do:
```rust
            let message_as_serde_value = serde_json::to_value(&m).unwrap().as_object().unwrap().values().next().unwrap().clone();
```
 This will take the serde::Value that you want. You can use it everywhere in this function instead of using the path var